server/src/user/middleware.ts

```diff
@@ -27,68 +31,220 @@ export class Middleware {
+                    forkJoin(
+                        this.service.getUser((authenticationToken.sub)),
+                        of(authenticationToken)
+                    )
+                ),
+                catchError((err: Error) => {
+                    if(err instanceof NotFoundError) {
+                        // authenticationToken refers to a user that doesn't exist, rethrow as unauthorized error
+                        throw new UnauthorizedError(`Invalid authenticationToken: ${err.message}`);
+                    }
+
+                    throw err;
+                }),
+            ).subscribe({
+                next: ([user, authenticationToken]) => {
+                    socket.mazenet = {
+                        sessionId: socket.id,
+                        userId: authenticationToken.sub
+                    };
 
-            socket.on('disconnect', () => {
-                this.service.onUserDisconnect(socket.mazenet!.sessionId);
+                    socket.on('disconnect', () => {
+                        this.service.onUserDisconnect(socket.mazenet!.sessionId);
+                    });
+                    return next();
+                },
+                error: (error: Error) => {
+                    GlobalLogger.error(
+                        'Socket failed authentication handshake on connect',
+                        {error}
+                    );
+
+                    return next(error);
+                }
             });
-            next();
         };
         this.router = Express.Router();
+
         this.router.use((req: Request, res: Response, next: Express.NextFunction) => {
             const socketData = (req.socket as Socket).mazenet;
             if(socketData) {
-                // this is a socketio connection
-                //TODO: do we need to validate that activeUser is valid for this user?
-                //TODO: standardize error handling
-                req.activeUser = this.service.getActiveUserFromSession(socketData!.sessionId);
-                if(!socketData!.user) {
-                    this.service.createUser({username: 'anonymous'}).subscribe((user) => {
-                        socketData!.user = user;
-                        req.user = user;
-                        return next();
-                    }, (error) => next(error));
-                } else {
-                    req.user = socketData!.user;
-                    return next();
+                // websocket doesn't need to validate JWT because it was validated when the connection was established
+                // NOTE: should we kill long-running websocket connections that have outlived their token lifespan?
+                req.userId = socketData.userId;
+                return next();
+            }
+            else {
+                if(!req.cookies.authenticationToken) {
+                    // create an anonymous user for this session
+                    return this.createGuest().subscribe({
```

Comment:
We are creating a new user for every session, which results in a new user being created every time a bot tries to hit some random URL like `/wp-admin`. I think the most sensible solution to this problem is to only create a guest account when accessing the root `/` endpoint--especially because it doesn't make a lot of sense to create a guest user when accessing the HTTP API (it should be an explicit action in that case). I don't think a normal user would hit any other endpoints and end up needing a guest user account?

Whether we should support and continue to build out out the HTTP API is a different discussion... I still believe in the idea of user-created mazenet bots that can use the API to like, post whatever content like news feeds and stuff, or like, external moderator bots and that kind of thing, and the socketio-router library makes the HTTP API basically free anyway so it's not like it's adding any significant amount of work on the server side.